flake.nix

```diff
@@ -1,68 +1,64 @@
 {
   inputs = {
-    flake-utils.url = "github:numtide/flake-utils";
-    haskellNix.url = "github:input-output-hk/haskell.nix";
-    nixpkgs.follows = "haskellNix/nixpkgs-unstable";
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-24.11";
+    systems.url = "github:nix-systems/default";
+    devenv.url = "github:cachix/devenv";
+    devenv.inputs.nixpkgs.follows = "nixpkgs";
   };
 
   nixConfig = {
-    allow-import-from-derivation = "true";
-    extra-substituters = [
-      "https://cache.iog.io"
-      "https://cache.zw3rk.com"
-    ];
-    extra-trusted-public-keys = [
-      "hydra.iohk.io:f/Ea+s+dFdN+3Y/G+FDgSq+a5NEWhJGzdjvKNGv0/EQ="
-      "loony-tools:pr9m4BkM/5/eSTZlkQyRt57Jz7OMBxNSUiMC4FkcNfk="
-    ];
+    extra-trusted-public-keys = "devenv.cachix.org-1:w1cLUi8dv3hnoSPGAuibQv+f9TZLr6cv/Hm9XgU50cw=";
+    extra-substituters = "https://devenv.cachix.org";
   };
 
-  outputs = inputs@{ self, flake-utils, haskellNix, nixpkgs }:
-    # https://input-output-hk.github.io/haskell.nix/tutorials/getting-started-flakes.html
-    flake-utils.lib.eachDefaultSystem (system:
-      let
-        overlays = [
-          haskellNix.overlay
-          (final: prev: {
-            hapistrano = final.haskell-nix.cabalProject' {
-              src = final.haskell-nix.haskellLib.cleanGit {
-                name = "hapistrano";
-                src = ./.;
-              };
-              # This is used by `nix develop .` to open a shell for use with
-              # `cabal`, `hlint` and `haskell-language-server`
-              shell.tools = {
-                cabal = {};
-                hlint = {};
-                haskell-language-server = {};
-              };
-              compiler-nix-name = "ghc966";
-            };
-          })
-        ];       
-        pkgs = import nixpkgs { inherit system overlays; inherit (haskellNix) config; };
-        flake = pkgs.hapistrano.flake { };
-      in rec {
-        apps = {
-          test = {
-            type = "app";
-            program = "${packages.test}/bin/test";
-          };
-        };
-        packages = {
-          default = flake.packages."hapistrano:exe:hap";
-          test = flake.packages."hapistrano:test:test".overrideAttrs (_: {
-            postFixup = ''
-              wrapProgram $out/bin/test \
-                --set PATH ${pkgs.lib.makeBinPath [
-                  pkgs.bash
-                  pkgs.coreutils
-                  pkgs.findutils
-                  pkgs.git
-                  pkgs.zsh
-                ]}
-            '';
-          });
+  outputs = {
+    self,
+    nixpkgs,
+    devenv,
+    systems,
+    ...
+  } @ inputs: let
```

Comment:
@DavidMazarro I'm wondering if the complexity of this file comes from the fact that we are using it for two different purposes. On one hand, it appears that we want a portable development environment, and on the other, we want to provide a Nix flake version of this project that can be properly referenced in other Nix projects. If that's the case, I'd recommend creating a standalone `devenv.nix` file for local development and leaving the current `flake.nix` with only the `packages` declaration; this should simplify some maintenance tasks as `devenv` is a higher-level abstraction over Nix flakes, making it more approachable for new comers.